src/components/UserPage/CreateTrip/CreateTrip.js

```diff
@@ -42,10 +63,59 @@ export const CreateTrip = ({ actions, toggleCreateTripModal }) => {
     });
   };
 
+  const searchDestination = query => {
+    if (query) {
+      placeServices.textSearch({ query }, data => {
+        const destinationDetail = {};
+
+        if (data[0]) {
+          destinationDetail.photos = data[0].photos.map(photo => {
+            return {
+              url: photo.getUrl(),
+              height: photo.height,
+              width: photo.width,
+            };
+          });
+          destinationDetail.name = data[0].name;
+          destinationDetail.geo = new firebase.firestore.GeoPoint(
+            data[0].geometry.location.lat(),
+            data[0].geometry.location.lng(),
+          );
+          destinationDetail.place_id = data[0].place_id;
+          destinationDetail.types = data[0].types;
+          destinationDetail.formatted_address = data[0].formatted_address;
+        }
+
+        setValues({
+          ...form,
+          destinations: [...form.destinations, destinationDetail],
+        });
```

Comment:
how does this set state function work? where is the ...form coming from?

Reply:
form is this: https://github.com/kevinlin505/goplan/pull/13/files#diff-a47ac536195e19af9d72d9ca63375f2aR42

Reply:
Basically, I am updating the destinations array.

Reply:
got it. makes a lot more sense knowing what form and destructure do